### PR TITLE
feat: aot runtime with revmc

### PR DESCRIPTION
**Pull request by winterjihwan**

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -26,6 +26,27 @@ jobs:
...
+        run: |
+          prefix=$(llvm-config-18 --prefix)
+          echo "LLVM_SYS_180_PREFIX=$prefix" >> $GITHUB_ENV
+
+      - name: Verify LLVM Installation
+        run: |
+          echo "Using LLVM version:"
+          llvm-config-18 --version
+          echo "LLVM_SYS_180_PREFIX=$LLVM_SYS_180_PREFIX"
+
```
Comment (djm07073) [inline .github/workflows/lint.yml:49]: move to seperate github actions

File: .gitignore [hunk 2]
```diff
@@ -16,6 +16,16 @@ Cargo.lock
 # MacOS
 .DS_Store
 
+# No static libraries (35MB+)
+lib*.a
+lib*.dylib
```
Comment (djm07073) [inline .gitignore:21]: delete it

File: Makefile [hunk 3]
```diff
@@ -1,4 +1,7 @@
-.PHONY: all build build-rust build-go test precompile
+.PHONY: all build build-rust build-go test precompile clean_sleddb
+
+HOME_DIR := $(HOME)
+SLEDDB_PATH := $(HOME_DIR)/rethmint/data
```
Comment (djm07073) [inline Makefile:4]: how about using name db not dat?

File: api/cron.go [hunk 4]
```diff
@@ -0,0 +1,25 @@
+package api
```
Comment (djm07073) [inline api/cron.go:1]: rename filname cron to aot_compiler

File: lib.go [hunk 5]
```diff
@@ -10,6 +10,23 @@ type VM struct {
...
+}
+
+func (compiler *Compiler) Start(kvstore api.KVStore) {
+	api.StartRoutine(compiler.Inner, kvstore)
+}
+
+func (compiler *Compiler) Destroy() {
+	api.ReleaseCompiler(compiler.Inner)
+}
+
```
Comment (djm07073) [inline lib.go:29]: use same db(kvstore) of vm store not the copy of db(kvstore). if vm store db is commited by tx, but compiler kvstore db is still not changed.

File: lib_test.go [hunk 6]
```diff
@@ -3,34 +3,49 @@ package revm_api_test
...
 	testutils.Faucet(kvStore, caller, big.NewInt(1000000000000))
 
 	return vm, kvStore, caller
 }
 
-func Test_e2e(t *testing.T) {
-	vm, kvStore, caller := setupTest(t)
-	// Deploy Test Contract
-	txData, err := hexutil.Decode(testca.TestBin)
+func Test_e2e_aot(t *testing.T) {
```
Comment (djm07073) [inline lib_test.go:45]: how about making two test case(use aot true / false)? just keep the `Test_e2e`  code and add `Test_e2e_aot` test on it

File: lib_test.go [hunk 7]
```diff
@@ -42,68 +57,29 @@ func Test_e2e(t *testing.T) {
...
-		GasUsed:     27766,
-		GasRefunded: 0,
-		Logs:        []types.Log{},
-		Output: types.Output{
-			DeployedAddress: common.Address{},
-			Output: []byte{
-				0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1,
-			},
-		},
-	}, queryRes)
```
Comment (djm07073) [inline lib_test.go:108]: reuse exeuction result assertion in `Test_e2e_aot`

**Review by djm07073 — CHANGES_REQUESTED** [review]

File: lib_test.go (hunk 7 above)
Comment (winterjihwan) [reply lib_test.go]: resolved

File: lib_test.go (hunk 6 above)
Comment (winterjihwan) [reply lib_test.go:45]: work in progress, getting cgo error at the moment

File: api/cron.go (hunk 4 above)
Comment (winterjihwan) [reply api/cron.go]: resolved

File: Makefile (hunk 3 above)
Comment (winterjihwan) [reply Makefile]: resolved

File: .github/workflows/lint.yml (hunk 1 above)
Comment (winterjihwan) [reply .github/workflows/lint.yml:49]: Is either build or release a better place for llvm configuration?

File: lib_test.go (hunk 6 above)
Comment (winterjihwan) [reply lib_test.go:45]: resolved

File: .github/workflows/lint.yml (hunk 1 above)
Comment (djm07073) [reply .github/workflows/lint.yml:49]: or make the new seperate github actions such as install.yml or llvm.yml

File: .github/workflows/lint.yml [hunk 8]
```diff
@@ -26,6 +26,27 @@ jobs:
...
+          sudo apt-get update
+          sudo apt-get install -y lsb-release wget software-properties-common
+          wget https://apt.llvm.org/llvm.sh
+          chmod +x llvm.sh
+          sudo ./llvm.sh 18
+
+      - name: Set LLVM_SYS_180_PREFIX
+        run: |
+          prefix=$(llvm-config-18 --prefix)
+          echo "LLVM_SYS_180_PREFIX=$prefix" >> $GITHUB_ENV
```
Comment (djm07073) [inline .github/workflows/lint.yml:42]: why put the log in the github env?

File: Makefile [hunk 9]
```diff
@@ -24,20 +24,51 @@ else
...
+lib-test:
+	make build-rust-debug
+	go clean -testcache
+	go test -v -run TestEofFibWithAOT
+
+go-test:
+	go clean -testcache
+	go test -v -run TestEofFibWithAOT
+
+clean-compile-sleddb:
```
Comment (djm07073) [inline Makefile:47]: clean-db? be short cmd. looks simple

File: Makefile [hunk 10]
```diff
@@ -24,20 +24,51 @@ else
...
 	cp librevm/bindings.h api
 
+lib-test:
+	make build-rust-debug
+	go clean -testcache
+	go test -v -run TestEofFibWithAOT
+
+go-test:
+	go clean -testcache
+	go test -v -run TestEofFibWithAOT
```
Comment (djm07073) [inline Makefile:45]: duplicated cmd

File: librevm/src/compiler/sleddb.rs [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    pub fn get(&self, key: K) -> Result<Option<IVec>> {
+        self.db.get(key).map_err(|e| eyre::Report::new(e))
+    }
+
+    pub fn key_iterator(&self) -> impl Iterator<Item = IVec> {
+        self.db.iter().keys().filter_map(|res| res.ok())
+    }
+}
```
Comment (djm07073) [inline librevm/src/compiler/sleddb.rs:51]: it seems ununsed method. delete unused method of sled db.

**Pull request merged**